### ISSUE TITLE
Sticky header on mobile when Admin Menu Tooltip is showing (Safari)

### ISSUE THREAD

**Issue by kelvinballoo** [google/site-kit-wp#8764]
## Bug Description

Under normal circumstances, it seems like the top header of the Site kit dashboard on mobile is not sticky.
When we have the tooltip of the audience segmentation on, then the header becomes sticky and the layout becomes weird.

## Steps to reproduce

1. Enable audience segmentation flag and analytics
2. On a mobile, on the banner for enabling groups, click on 'Maybe later' 
3. This will reveal a tooltip and scroll the page

## Screenshots
Video 1: without tooltip:

https://github.com/google/site-kit-wp/assets/125576926/4d06c090-d335-446a-ab24-dee2f6124206

Video 2: With tooltip on:
https://github.com/google/site-kit-wp/assets/125576926/1d5dd6aa-80c6-4580-a625-7620c56cc4b6

Weird layout:
<img width="268" alt="Screenshot 2024-05-29 at 11 33 52" src="https://github.com/google/site-kit-wp/assets/125576926/bc63c098-560a-4168-99f3-45af470b9f87">

## Additional Context

 - PHP Version: 8.0
 - OS: 17.4.1
 - Browser: Safari
 - Plugin Version: 1.127.0
 - Device: iPhone 15Pro Max

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When the Admin Menu Tooltip is showing (e.g. having run through the **steps to reproduce**, or showing it via another means for example clicking "Maybe later" on the Key Metrics Setup CTA), the Site Kit header should not be sticky.
* Specifically this is known to need fixing on Safari while it works on Chrome and Firefox ([tested on a Pixel 6a](https://github.com/google/site-kit-wp/issues/8764#issuecomment-2147367344)). Latest Edge should also be checked and fixed as necessary.

## Implementation Brief

* There is a [POC](https://github.com/google/site-kit-wp/pull/8867) implementation, it can be used with a bit of polishing where needed
* [ ] Implement a body class for when `JoyrideTooltip` component is shown, and remove it, when tour is ended (`EVENTS.TOUR_START/EVENTS.TOUR_END`)
  * You can use `googlesitekit-showing-tooltip`
  * So it can behave same as `googlesitekit-showing-feature-tour` which already has this implementation 
* [ ] Update `assets/sass/components/global/_googlesitekit-header.scss`, `assets/sass/components/global/_googlesitekit-entity-header.scss`, `assets/sass/components/global/_googlesitekit-dashboard-navigation.scss`:
  * Include new class selector, together with existing `body.googlesitekit-showing-feature-tour &` selector, so `static` position is applied to the header and menu

### Test Coverage

* No changes expected

## QA Brief

* Go through steps to reproduce
* Verify that header is not sticky when admin menu tooltip is showing

## Changelog entry

* Fix the problem with sticky header when the admin menu tooltip is shown.

**Comment by techanvil** [comment]
Thanks for raising this @kelvinballoo, it's a good spot.

I've found the issue isn't occurring on Chrome and Firefox on a Pixel 6a, while it is reproducible for me using Safari on a Mac.

Here's a screencast of Chrome:

https://github.com/google/site-kit-wp/assets/18395600/f38e2154-9145-442b-b8c2-3cb3f408fcc6

It's also worth noting it's a general issue that applies to all cases where we show a tooltip pointing to the admin menu, so I've removed this from the Audience Segmentation epic, and unassigned it from Squad M, as either squad can pick it up.
